frontends/p4/tableKeyNames.cpp

```diff
@@ -32,10 +38,16 @@ class KeyNameGenerator : public Inspector {
 
     void postorder(const IR::Expression* expression) override { error(expression); }
 
+    /** Compute a name annotation for @expression.  Eg. `@name("foo.bar")` for
```

Comment:
To me, the fact that some of these comments are not accurate suggests that we need to either simplify the code, or add documentation so anyone can easily understand it.

I think having Doxygen comments for most methods is reasonable for uniformity, even if the meaning is clear from the name and type signature.

Finally, if a PhD-level engineer stays up past midnight to add documentation to code that somebody else wrote, it seems only polite to avoid characterizing their effort as "practically useless" ;-)